### PR TITLE
Improve the commons with full API response capture 

### PR DESCRIPTION
**Pull request by camillaol**
**TLDR**
This PR represents a 96% rewrite of the `common.go` structs for capturing data from the Forge API; this is in the interest of _directly matching_ the structure of responses from the Forge API. 

**Context**
When I wrote the initial version, I did not have a good understanding of how to do this correctly (first timer here 😄!). I tried to pick and choose the fields that we'd capture, rather than matching the API responses field for field.

This rewrite represents a direct match to the JSON structure given by the Forge API, enabling a user to pick and choose the fields they want to keep at the application level rather than being given any restrictions at the `forge-api-go-client` level. 

**File by file changes**
* `dm/common.go` - 96% rewrite, as mentioned above
* `dm/hubs.go`, `dm/projects.go`, `dm/folders.go`, `dm/items.go` - minor edits to update struct reference for data capture
* `dm/hub_test.go`, `dm.project_test.go`, `dm.folder_test.go` - minor edits to change local package import to full github path import

**Testing**
In addition to passing unit tests, I have pulled the latest version of this branch into an active Outer Labs project for testing, to full success. There **are** failing tests that persist: they are bucket/object tests which are part of a previous set of work which I am punting on for now.

### CODE REVIEW

File: dm/common.go [hunk 1]
```diff
@@ -1,111 +1,101 @@
...
+	Relationships 				*Relationships 		`json:"relationships, omitempty"`
+	Links 					*Links 			`json:"links, omitempty"`
 }
 
-type JsonAPI struct {
-	Version 	string 		`json:"version, omitempty"`
+type Attributes struct {
+	Name      				string 			`json:"name"`
+	Extension 				Extension 		`json:"extension"`
+	Region	  				*string 		`json:"region, omitempty"`
```
Comment (monicatie) [inline dm/common.go:40]: curious what makes this one a `*string` and `Name` above a `string`?

**Review by monicatie — COMMENTED** [review]
Will this refactor work with Compass (and Blueprint)? Or will we need to update code on the other apps?

File: dm/common.go (hunk 1 above)
Comment (camillaol) [reply dm/common.go:40]: The fields with a pointer relate to the `optional` or "omitempty" nature of them. Based on my understanding from [this article](https://www.sohamkamani.com/blog/golang/2018-07-19-golang-omitempty/), the pointer allows the existence of a `null` value for a given field.

**Comment by camillaol** [comment]
Other apps will need to be updated. I have made the changes in Compass already. I'd be happy to help in Blueprint, too!

File: dm/common.go (hunk 1 above)
Comment (monicatie) [reply dm/common.go:40]: ooo cool cool
I see, so name isn't optional so its not a pointer?

**Comment by monicatie** [comment]
> Other apps will need to be updated. I have made the changes in Compass already. I'd be happy to help in Blueprint, too!

if you could add me on a MR for blueprint too that would be great!

**Comment by camillaol** [comment]
Blueprint MR is up in Gitlab. Merging here!
